app-modules/engagement/src/Models/EngagementFile.php

```diff
@@ -73,7 +73,7 @@ public function students(): MorphToMany
             table: 'engagement_file_entities',
             foreignPivotKey: 'engagement_file_id',
             relatedPivotKey: 'entity_id',
-            relation: 'student',
+            relation: 'engagementFiles',
```

Comment:
I think this should remain `student`

Reply:
It seems to break when I set it as `student`

Reply:
That's odd 🤔

Reply:
I'll continue to look into it more. But this change has it working now